### PR TITLE
Add audit operation in chooseCandidateReplica for handling error

### LINKED ISSUE #1057: Add audit operation in chooseCandidateReplica for handling error
I want to catch an error from `replica.CanReplicateFrom`.
https://github.com/github/orchestrator/blob/master/go/inst/instance_topology.go#L2182

And output this using `AuditOperation` like this.
```Go
if canReplicate, err := replica.CanReplicateFrom(candidateReplica); !canReplicate {
    // lost due to inability to replicate
    cannotReplicateReplicas = append(cannotReplicateReplicas, replica)
    if err != nil {
        AuditOperation("chose-candidate-replica", &replica.Key, err.Error())
    }
}
```

### PR DESCRIPTION
**Pull request by rluisr**
Fixes: #1057

### CODE REVIEW

**Review by shlomi-noach — CHANGES_REQUESTED** [review]
Thank you for the PR!
`AuditOperation()` is reserved to listing complete operations (like relocate, start/stop replication, etc.) once they are complete, and not for general purpose logging.

Please use `log.Errorf(...)` to log this error.

**Comment by rluisr** [comment]
@shlomi-noach Thanks review.
I changed. Please review again.

**Comment by shlomi-noach** [comment]
@rluisr please note the complication error.

**Comment by rluisr** [comment]
@shlomi-noach sorry. fixed.

File: go/inst/instance_topology.go [hunk 1]
```diff
@@ -2179,9 +2179,12 @@ func chooseCandidateReplica(replicas [](*Instance)) (candidateReplica *Instance,
 	replicas = RemoveInstance(replicas, &candidateReplica.Key)
 	for _, replica := range replicas {
 		replica := replica
-		if canReplicate, _ := replica.CanReplicateFrom(candidateReplica); !canReplicate {
+		if canReplicate, err := replica.CanReplicateFrom(candidateReplica); !canReplicate {
 			// lost due to inability to replicate
 			cannotReplicateReplicas = append(cannotReplicateReplicas, replica)
+			if err != nil {
+				_ = log.Errorf("This replica cannot replicate.\nreplica: %v\nerror: %v", replica, err)
```
Comment (shlomi-noach) [inline go/inst/instance_topology.go]: suggestion: one liner log, same as everywhere else. Instance key is enough, no need to for entire replica struct.
```suggestion
				log.Errorf("chooseCandidateReplica(): error checking CanReplicateFrom(). replica: %v; error: %v", replica.Key, err)
```

**Review by shlomi-noach — APPROVED** [review]

**Pull request merged**